internal/runner/group_executor.go

```diff
@@ -43,32 +43,60 @@ type DefaultGroupExecutor struct {
 // groupNotificationFunc is a function type for sending group notifications
 type groupNotificationFunc func(group *runnertypes.GroupSpec, result *groupExecutionResult, duration time.Duration)
 
-// NewDefaultGroupExecutor creates a new DefaultGroupExecutor
+// NewDefaultGroupExecutor creates a new DefaultGroupExecutor with the specified
+// configuration and optional settings.
 func NewDefaultGroupExecutor(
 	executor executor.CommandExecutor,
 	config *runnertypes.ConfigSpec,
 	validator security.ValidatorInterface,
 	verificationManager verification.ManagerInterface,
 	resourceManager resource.ResourceManager,
 	runID string,
-	notificationFunc groupNotificationFunc,
-	isDryRun bool,
-	dryRunDetailLevel resource.DetailLevel,
-	dryRunShowSensitive bool,
-	keepTempDirs bool,
+	options ...GroupExecutorOption,
 ) *DefaultGroupExecutor {
+	// Input validation
+	if config == nil {
+		panic("NewDefaultGroupExecutor: config cannot be nil")
+	}
+	if resourceManager == nil {
+		panic("NewDefaultGroupExecutor: resourceManager cannot be nil")
+	}
+	if runID == "" {
+		panic("NewDefaultGroupExecutor: runID cannot be empty")
+	}
+
+	// Apply options
+	opts := defaultGroupExecutorOptions()
+	for _, opt := range options {
+		if opt != nil {
+			opt(opts)
+		}
+	}
+
+	// Extract dry-run settings
+	isDryRun := opts.dryRunOptions != nil
+	var detailLevel resource.DetailLevel
```

Comment:
This is only used for dry-run mode. Might be better to change the variable name like dryRunDetailLevel